optax/_src/transform_test.py

```diff
@@ -202,6 +202,16 @@ def test_centralize(self, inputs, outputs):
     centralized_inputs, _ = centralizer.update(inputs, None)
     chex.assert_tree_all_close(centralized_inputs, outputs)
 
+  def test_sherman_morrisson(self):
+    x = 1
+    a_inv = jnp.eye(3) * 1.0
+    u = v = jnp.ones(3)
+
+    ref = jnp.linalg.inv(a_inv + x * jnp.outer(u, v))
+
+    a_inv = transform.sherman_morrison(a_inv, u, v)
+    assert jnp.allclose(a_inv, ref)
+
```

Comment:
Please add a test for `scale_by_newton` that tests both `init` and `update`. The update function can be tested e.g. by comparing an analytic calculation against a step taken with the code.